src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/MSBuildShellOutNuGetProject.cs

```diff
@@ -56,6 +57,9 @@ public class MSBuildShellOutNuGetProject : NuGetProject, INuGetIntegratedProject
 
         private const string MSBuildExe = "msbuild.exe";
 
+        private const string NuGetSubKey = @"Software\NuGet";
+        private const string EnableRegistryKey = "EnableMSBuildShellOutNuGetProject";
```

Comment:
And also add this registry key in visualstudio.ui Constants class to have all these keys at one place?